CMakeLists.txt

```diff
@@ -37,8 +37,21 @@ endif()
 # Boost package
 ###############################################################################
 
+set(Boost_COMPONENTS coroutine context system thread)
+
+# If on Windows, rather than be annoyed by constant failure to find DLL errors
+# during test debugging simply force all static linking of Boost libraries
+if(WIN32 AND NOT MINGW)
+  set(Boost_USE_STATIC_LIBS ON)
+else()
+  set(Boost_COMPONENTS ${Boost_COMPONENTS} unit_test_framework)
```

Comment:
It's more that Boost.Test has a long standing bug whereby header only usage doesn't work on Mingw, so you always have to link against the dynamic libraries because the static libraries are useless on Mingw. You want a note to that effect?